internal/topology/variables/clusterclass_variable_validation.go

```diff
@@ -29,6 +29,12 @@ import (
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
+const (
+	// builtinsName is the name of the builtin variable.
+	// NOTE: User-defined variables are not allowed to start with "builtin".
```

Comment:
I think we currently define the rule as variables are not allowed to be "builtin". My understanding is that builtin124 is a valid variable name, but builtin.clustername is not (because we reject any variable with a '.'

Reply:
Remove d the note entirely (and in the other places where the const is defined).